### PR TITLE
NAMD cpu Test

### PR DESCRIPTION
**Pull request by Lewih**
CPU test for NAMD.
Missing reference for genius.
Executed on hydra successfully  with not smp module NAMD/2.14-foss-2019b-mpi.

### CODE REVIEW

**Comment by Lewih** [comment]
@smoors may u test on hydra?

**Comment by Lewih** [comment]
@stevenvdb On genius it does not execute reliably, any idea? 

`[     FAIL ] (1/6) Namd_NotSMP_CPUTest %num_nodes=4 @genius:single-node+builtin 

[     FAIL ] (2/6) Namd_NotSMP_CPUTest %num_nodes=2 @genius:single-node+builtin 

[     FAIL ] (3/6) Namd_NotSMP_CPUTest %num_nodes=1 @genius:single-node+builtin

[     FAIL ] (4/6) Namd_SMP_CPUTest %num_nodes=4 @genius:single-node+builtin 

[       OK ] (5/6) Namd_SMP_CPUTest %num_nodes=2 @genius:single-node+builtin 

[       OK ] (6/6) Namd_SMP_CPUTest %num_nodes=1 @genius:single-node+builtin 
`

**Comment by stevenvdb** [comment]
@Lewih See https://github.com/Lewih/vsc-test-suite/pull/1/files

File: tests/apps/namd/namd_test.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        if self.current_system.name == 'leibniz':
+            self.num_cpus_per_task = 28
+            self.modules = ['NAMD/2.14-verbs']
+        elif self.current_system.name == 'vaughan':
+            self.num_cpus_per_task = 64
+            self.modules = ['NAMD/2.14-verbs']
+        elif self.current_system.name == 'hydra':
+            self.num_tasks = 40 * self.num_tasks
+            self.modules = ['NAMD/2.14-foss-2019b-mpi']
+            self.job.options = ["--partition=skylake,skylake_mpi",
```
Comment (smoors) [inline tests/apps/namd/namd_test.py:158]: for multi-node jobs you need to use skylake_mpi to get infiniband interconnect
```suggestion
            self.job.options = ["--partition=skylake_mpi",
```

File: tests/apps/namd/namd_test.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+        # VSC specific config
+        if self.current_system.name == 'leibniz':
+            self.num_cpus_per_task = 28
+            self.modules = ['NAMD/2.14-verbs']
+        elif self.current_system.name == 'vaughan':
+            self.num_cpus_per_task = 64
+            self.modules = ['NAMD/2.14-verbs']
+        elif self.current_system.name == 'hydra':
+            self.num_tasks = 40 * self.num_tasks
```
Comment (smoors) [inline tests/apps/namd/namd_test.py:156]: you should also specify num_tasks_per_node to ensure full nodes are used:
```suggestion
            self.num_tasks = 40 * self.num_tasks
            self.num_tasks_per_node = 40
```

File: tests/apps/namd/namd_test.py (hunk 1 above)
Comment (Lewih) [reply tests/apps/namd/namd_test.py:158]: LGTM

**Comment by Lewih** [comment]
TODO
on Vaughan the test shows problems

**Comment by Lewih** [comment]
@stevenvdb,  I just noticed `NAMD/2.14-foss-2019b-mpi` module is defined for both the not SMP **and SMP tests**, do you have also a SMP build installed? Otherwise I will remove the gaussian entry for the SMP test in order to not repeat 2 times the same test.

File: tests/apps/namd/namd_test.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            },
+            '2': {
+                'leibniz:single-node': {'days_ns': (0.1782715, None, 0.05, 'days/ns')},
+                'vaughan:single-node': {'days_ns': (0.09856985, None, 0.05, 'days/ns')},
+                'hydra:single-node': {'days_ns': (0.1099565, None, 0.05, 'days/ns')},
+            },
+            '4': {
+                'leibniz:single-node': {'days_ns': (1.05726, None, 0.05, 'days/ns')},
+                'vaughan:single-node': {'days_ns': (0.5438339, None, 0.05, 'days/ns')},
+                'hydra:single-node': {'days_ns': (0.5427335, None, 0.05, 'days/ns')},
```
Comment (smoors) [inline tests/apps/namd/namd_test.py:135]: it seems that running in 4 nodes tanks efficiency, maybe we should drop it?

Comment (Lewih) [reply tests/apps/namd/namd_test.py:135]: On 4 nodes it is actually a different test, uses a different simulation, a bigger one.

Comment (smoors) [reply tests/apps/namd/namd_test.py:135]: ah ok, no problem then

**Comment by Lewih** [comment]
This works on all the systems.

I have noticed some performance degradation on hydra which does not meet the reference in this run.
```
 Namd_NotSMP_CPUTest %num_nodes=2
   - builtin
      * num_tasks: 80
      * days_ns: 0.16945 days/ns
```
I assume we will have a better idea about the references while the tests will be run regularly.

**Comment by smoors** [comment]
> I have noticed some performance degradation on hydra which does not meet the reference today.

interesting. one reason could be that the default output and prefix dirs in `run.sh` are in $HOME, which is a quite slow NFS mount.
that's why I run with options `--prefix $VSC_SCRATCH/reframe-vsc --output $VSC_SCRATCH/reframe-vsc`. maybe we should make that the default?

but that should not hold back from merging this PR.

**Review by smoors — APPROVED** [review]
lgtm

**Pull request merged**